server/src/main/java/org/elasticsearch/cluster/routing/RoutingTable.java

```diff
@@ -331,7 +316,9 @@ public static Diff<RoutingTable> readDiffFrom(StreamInput in) throws IOException
 
     public static RoutingTable readFrom(StreamInput in) throws IOException {
         Builder builder = new Builder();
-        builder.version = in.readLong();
+        if (in.getTransportVersion().before(TransportVersions.ROUTING_TABLE_VERSION_REMOVED)) {
+            in.readLong();
```

Comment:
probably going to save some digging later to comment on this:

```suggestion
            in.readLong(); // previously 'version', now unused
```